crypto-pgpainless/src/main/kotlin/dev/msfjarvis/aps/crypto/PGPainlessCryptoHandler.kt

```diff
@@ -64,24 +64,33 @@ public class PGPainlessCryptoHandler @Inject constructor() : CryptoHandler<PGPKe
     outputStream: OutputStream,
   ): Result<Unit, CryptoHandlerException> =
     runCatching {
+        if (keys.isEmpty()) throw NoKeysProvided("No keys provided for encryption")
         val armoredKeys = keys.map { key -> key.contents.decodeToString() }
         val pubKeysStream = ByteArrayInputStream(armoredKeys.joinToString("\n").toByteArray())
         val publicKeyRingCollection =
-          pubKeysStream.use {
-            ArmoredInputStream(it).use { armoredInputStream ->
-              PGPainless.readKeyRing().publicKeyRingCollection(armoredInputStream)
-            }
+          pubKeysStream.use { PGPainless.readKeyRing().publicKeyRingCollection(pubKeysStream) }
+        val encryptionOptions =
+          EncryptionOptions.encryptCommunications()
```

Comment:
Here you should just use `new EncryptionOptions()` (or the Kotlin equivalent of that), otherwise you limit yourself to keys with key flag `ENCRYPT_COMMS`, ignoring those with `ENCRYPT_STORAGE`.

My suggested code will make use of either.